File: ts/tests/add-team.test.ts
import { test, expect, chromium } from '@playwright/test'

test('has title', async () => {
  const browser = await chromium.launch({ slowMo: 1000 })
  const page = await browser.newPage()

  // Reset database
  await page.goto('/reset_db')
  const proceedButton = page.locator("button:has-text('proceed')")
  await proceedButton.click()

  // Create a new team
  await page.goto('/add_team')
  const nameInput = page.locator('input[name="name"]')
  const team_name = 'my team'
  await nameInput.fill(team_name)
  await page.click("text='Add'")

  // Check the team has been created
  await page.goto("/teams")
  await page.isVisible(`td:has-text('${team_name}')`)
})

File: java/src/test/java/info/dmerej/AddTeamTest.java
package info.dmerej;

import com.microsoft.playwright.Browser;
import com.microsoft.playwright.BrowserType;
import com.microsoft.playwright.Playwright;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertTrue;

public class AddTeamTest {
    @Test
    void test_add_team() {
        // Use playwright driver to get a browser and open a new page
        var playwright = Playwright.create();
        var launchOptions = new BrowserType.LaunchOptions().setHeadless(false)
            .setSlowMo(1000); // Remove this when you're done debugging
        var browser = playwright.chromium().launch(launchOptions);

        // Set base URL for the new context
        var contextOptions = new Browser.NewContextOptions();
        // TODO: fix the URL
        contextOptions.setBaseURL("https://<letter>.<group>.hr.dmerej.info");
        var context = browser.newContext(contextOptions);

        var page = context.newPage();

        // Reset database
        page.navigate("/reset_db");
        var proceedButton = page.locator("button:has-text('proceed')");
        proceedButton.click();
        page.navigate("/");

        // Add a new team
        page.navigate("/add_team");
        var nameInput = page.locator("input[name=\"name\"]");
        var teamName = "my team";
        nameInput.fill(teamName);
        page.click("text='Add'");

        // Check that the team has been added
        page.navigate("/teams");

        // Check the new team is there
        String selector = String.format("td:has-text('%s')", teamName);
        var isVisible = page.isVisible(selector);
        assertTrue(isVisible);
    }
}
